packages/coil-extension/src/background/services/multipleInstanceDetector.ts

```diff
@@ -0,0 +1,135 @@
+import { inject, injectable } from 'inversify'
+
+import {
+  BrowserType,
+  ExtensionInstance,
+  EXTENSIONS
+} from '../consts/extensionInstances'
+import * as tokens from '../../types/tokens'
+
+export type WextApiSubset = {
+  permissions: {
+    contains: typeof chrome.permissions.contains
+  }
+  notifications: {
+    create: typeof chrome.notifications.create
+  }
+  tabs: {
+    create: typeof chrome.tabs.create
+  }
+  runtime: {
+    onMessageExternal: {
+      addListener: typeof chrome.runtime.onMessageExternal.addListener
+    }
+    lastError: typeof chrome.runtime.lastError
+    sendMessage: typeof chrome.runtime.sendMessage
+  }
+}
+
+export interface CheckActiveResponse {
+  active: boolean
+}
+
+export interface ExtensionMessage {
+  command: string
+}
+
+@injectable()
+export class MultipleInstanceDetector {
+  constructor(
+    private navigator: Navigator,
+    @inject(tokens.WextApi) private wextApi: WextApiSubset
+  ) {}
+
+  static getBrowserType(userAgent: string): BrowserType | null {
+    if (userAgent.includes('Firefox/')) {
+      return 'firefox'
+    } else if (userAgent.includes('Edg/')) {
+      return 'edge'
+    } else if (userAgent.includes('Chrome/')) {
+      return 'chrome'
+    } else {
+      return null
+    }
+  }
+
+  async detectOtherInstances() {
+    const browserType = MultipleInstanceDetector.getBrowserType(
+      this.navigator.userAgent
+    )
+    if (!browserType) {
+      return
+    }
+
+    const extensions = EXTENSIONS[browserType]
+    if (extensions.length <= 0) return
+
+    for (const extension of extensions) {
+      const active = await this.checkExtensionIsActive(extension)
+      if (active) {
+        this.showNotificationIfHavePerms(extension)
+        this.openTabForUserToManageExtensions(browserType)
+        break
+      }
+    }
+  }
+
+  private openTabForUserToManageExtensions(browserType: BrowserType) {
+    const mapping: Record<BrowserType, string> = {
+      chrome: 'chrome://extensions',
+      edge: 'edge://extensions',
+      firefox: 'about:debugging'
+    }
+    const url = mapping[browserType]
+    this.wextApi.tabs.create({ url })
+  }
+
+  showNotificationIfHavePerms(extensionData: ExtensionInstance) {
+    this.wextApi.permissions.contains(
+      {
+        permissions: ['notifications']
+      },
+      havePerm => {
+        if (havePerm) {
+          const notificationId = Date.now().toString()
+          const notification = {
+            contextMessage: 'There is already an active coil extension',
+            isClickable: false,
+            type: 'basic',
+            iconUrl: 'https://tinypng.com/images/example-shrunk.png',
```

Comment:
What is this?  Looks like a panda, we should use the Coil icon
https://tinypng.com/images/example-shrunk.png